pkg/syncer/cnsoperator/util/util.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/logger"
+)
+
+// GetVolumeID gets the volume ID from the PV that is bound to PVC by pvcName
+func GetVolumeID(ctx context.Context, client client.Client, pvcName string, namespace string) (string, error) {
+	log := logger.GetLogger(ctx)
+	// Get PVC by pvcName from namespace
+	pvc := &v1.PersistentVolumeClaim{}
+	err := client.Get(ctx, k8stypes.NamespacedName{Name: pvcName, Namespace: namespace}, pvc)
```

Comment:
Does this fetch object from API server? If yes, then we are hitting API server twice in this method.
Can this be optimized to use shared informer's PV/PVC listers? If it's not found using PVCLister, then we query the API server.

Reply:
I agree with this improvement. However I am creating a github issue for adding informers.

Reply:
Issue number 599

Reply:
Can you add a TODO here with a link to the issue?

Reply:
Done, added